src/coreclr/src/jit/morph.cpp

```diff
@@ -13324,6 +13324,60 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
 
                 /* No match - exit */
             }
+
+            // Skip optimization if non-NEG operand is constant.
+            // Both op1 and op2 are not constant because it was already checked above.
+            if (opts.OptimizationEnabled() &&
+                (((op1->gtFlags & GTF_EXCEPT) == 0) || ((op2->gtFlags & GTF_EXCEPT) == 0)))
+            {
+                // a - -b = > a + b
+                // SUB(a, (NEG(b)) => ADD(a, b)
+
+                if (!op1->OperIs(GT_NEG) && op2->OperIs(GT_NEG))
+                {
+                    // tree: SUB
+                    // op1: a
+                    // op2: NEG
+                    // op2Child: b
+
+                    GenTree* op2Child = op2->AsOp()->gtOp1; // b
+                    oper              = GT_ADD;
+                    tree->SetOper(oper);
```

Comment:
You will want to preserve the ValueNumber for this operation:

`tree->SetOper(oper, GewNtree::PRESERVE_VN));`

Alternatively do not perform this optimization, in the CSE phase:
(Above)
 `if (opts.OptimizationEnabled() && fgGlobalMorph &&`

Reply:
Thanks Brian. With this changes, the VN assert is gone.